frontend/__tests__/storylines.js

```diff
@@ -540,36 +561,175 @@ describe("Storylines", () => {
+        stubStoryline.state.global = getFooEqualBarState();
+        expect(stubStoryline.testPropositionalCondition({
+          _type: "propositional_condition",
+          boolean_operator: "OR",
+          conditions: [
+            {
+              _type: "atomic_condition",
+              lhs: buildState(['global', 'foo']),
+              operator: "==",
+              rhs: "nope"
+            },
+            {
+              _type: "atomic_condition",
+              lhs: true,
+              operator: "==",
+              rhs: true
+            }
+        ]})).toBeTruthy();
+      });
+
+      it("should return true when using OR and all conditions pass", () => {
+        stubStoryline.state.global = getFooEqualBarState();
+        expect(stubStoryline.testPropositionalCondition({
+          _type: "propositional_condition",
+          boolean_operator: "OR",
+          conditions: [
+            {
+              _type: "atomic_condition",
+              lhs: buildState(['global', 'foo']),
+              operator: "==",
+              rhs: "bar"
+            },
+            {
+              _type: "atomic_condition",
+              lhs: true,
+              operator: "==",
+              rhs: true
+            }
+        ]})).toBeTruthy();
+      });
+
+      it("should return false when using OR and all conditions fails", () => {
+        stubStoryline.state.global = getFooEqualBarState();
+        expect(stubStoryline.testPropositionalCondition({
+          _type: "propositional_condition",
+          boolean_operator: "OR",
+          conditions: [
+            {
+              _type: "atomic_condition",
+              lhs: buildState(['global', 'foo']),
+              operator: "==",
+              rhs: "nope"
+            },
+            {
+              _type: "atomic_condition",
+              lhs: true,
+              operator: "==",
+              rhs: false
+            }
+        ]})).toBeFalsy();
+      });
+
+      it("should fail on invalid boolean operator", () => {
+        stubStoryline.state.global = getFooEqualBarState();
+        expect(() => stubStoryline.testPropositionalCondition({
+          _type: "propositional_condition",
+          boolean_operator: "XOR",
+          conditions: []})).toThrow(/Invalid boolean operator/i);
+      });
+    });
+
+    describe("testCondition()", () => {
+      it("should recursively evaluate conditions (to true)", () => {
+        stubStoryline.state.global = getFooEqualBarState();
+        expect(stubStoryline.testCondition({
+          _type: "propositional_condition",
+          boolean_operator: "AND",
+          conditions: [
+            {
+              _type: "atomic_condition",
+              lhs: buildState(['global', 'foo']),
+              operator: "==",
+              rhs: "bar"
+            },
+            {
+              _type: "propositional_condition",
+              boolean_operator: "OR",
```

Comment:
This is testing lazy evaluation, not just recursive evaluation, because there is only one condition in this 'OR'

Reply:
It is recursive too, since it's nested in an AND